### PR TITLE
Grauwolf Sensor Probe Projectile

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Added a grauwolf probe projectile, can reveal things in a radius, does not hit anything and possess no damage, basically a minigame to extend sensor range.

Sprites made and supplied by Caelphon.

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!wip

File: code/modules/overmap/ship_weaponry/weaponry/grauwolf_probe.dm [hunk 1]
```diff
@@ -0,0 +1,105 @@
+/obj/item/ship_ammunition/grauwolf_probe
+	name = "grauwolf sensor probe"
+	desc = "A sensor probe, used to illuminate the area ahead using the dataling."
```
Comment (Generalcamo) [inline code/modules/overmap/ship_weaponry/weaponry/grauwolf_probe.dm:3]:
```suggestion
	desc = "A gun-launched sensor probe, used to scan the area ahead using the datalink."
```

**Review by Generalcamo — CHANGES_REQUESTED** [review]

**Comment by FluffyGhoster** [comment]
should be good, so !review

**Review by Alberyk — APPROVED** [review]

File: code/modules/overmap/ship_weaponry/weaponry/grauwolf_probe.dm [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	P.range = range
+	if(istype(origin, /obj/effect/overmap/visitable/ship))
+		var/obj/effect/overmap/visitable/ship/S = origin
+		P.dir = S.dir
+	P.icon_state = overmap_icon_state
+	P.speed = 32
+	P.entry_target = entry_point
+	forceMove(P)
+	log_and_message_admins("A projectile ([name]) has entered the Overmap! (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[P.x];Y=[P.y];Z=[P.z]'>JMP</a>)")
+	return TRUE
```
Comment (NonQueueingMatt) [inline code/modules/overmap/ship_weaponry/weaponry/grauwolf_probe.dm:33]: Don't really like this copypaste here. Whatever changes here should just be integrated into the parent proc if possible.

File: code/modules/overmap/ship_weaponry/weaponry/grauwolf_probe.dm [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	. = ..()
+
+/obj/effect/overmap/projectile/probe/Initialize(var/maploading, var/sx, var/sy, var/obj/effect/overmap/visitable/shooter)
+	origin = shooter
+	. = ..(maploading, sx, sy, shooter)
+
+/obj/effect/overmap/projectile/probe/Destroy()
+	var/obj/effect/overmap/visitable/ship/ship = origin
+
+	remove_contacts:
```
Comment (NonQueueingMatt) [inline code/modules/overmap/ship_weaponry/weaponry/grauwolf_probe.dm:86]: Origin needs to be nulled out here.

File: code/modules/overmap/ship_weaponry/weaponry/grauwolf_probe.dm [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+		var/obj/effect/overmap/visitable/ship/S = origin
+		P.dir = S.dir
+	P.icon_state = overmap_icon_state
+	P.speed = 32
+	P.entry_target = entry_point
+	forceMove(P)
+	log_and_message_admins("A projectile ([name]) has entered the Overmap! (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[P.x];Y=[P.y];Z=[P.z]'>JMP</a>)")
+	return TRUE
+/obj/effect/overmap/projectile/probe
+	var/obj/effect/overmap/visitable/origin = null
```
Comment (NonQueueingMatt) [inline code/modules/overmap/ship_weaponry/weaponry/grauwolf_probe.dm:35]: You don't need to redefine this. Already present in `ammunition.origin`

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**